include/sys/onoff_delayed.h

```diff
@@ -0,0 +1,185 @@
+ * on-off service API is invoked on the service.
+ *
+ * This function should never be invoked by clients of an on-off
+ * service.
+ *
+ * @param mgr the manager to be initialized.
+ *
+ * @param transitions pointer to a structure providing transition
+ * functions.  The referenced object must persist as long as the
+ * manager can be referenced.
+ *
+ * @param startup_time Service startup time + maximum interrupt latency given in
+ * system ticks. Start request is scheduled %p startup_time earlier to be
+ * ready no later than at requested time.
+ *
+ * @retval 0 on success.
+ * @retval -EINVAL if parameters are invalid.
+ */
+static inline int onoff_delayed_manager_init(struct onoff_delayed_manager *mgr,
+				const struct onoff_transitions *transitions,
+				uint32_t startup_time)
+{
+	mgr->startup_time = startup_time;
+
+	return onoff_manager_init(&mgr->mgr, transitions);
+}
+
+/**
+ * @brief Request on-off service to be in on state at specified time.
+ *
+ * Note that K_NO_WAIT or absolute timeout which is in the past results in
+ * immediate request and it is equivalent of calling on-off manager directly.
+ *
+ * @param mgr the manager that will be used.
+ *
+ * @param cli a non-null pointer to client state providing
+ * instructions on synchronous expectations and how to notify the
+ * client when the request completes.  Behavior is undefined if client
+ * passes a pointer object associated with an incomplete service
+ * operation.
+ *
+ * @param delay relative delay or absolute time at which service must be in on
+ * state.
+ *
+ * @retval 0 if successfully scheduled.
+ * @retval non-negative if delay is K_NO_WAIT or absolute time in the past and
+ * on-off manager was requested directly. See @ref onoff_request for details.
+ * @retval -EINVAL if the parameters are invalid.
+ * @retval negative on error returned by @ref onoff_request. See
+ * @ref onoff_request for details.
+ */
+int onoff_delayed_request(struct onoff_delayed_manager *mgr,
+			  struct onoff_delayed_client *cli,
+			  k_timeout_t delay);
+
+/**
+ * @brief Release a reserved use of an on-off service.
+ *
+ * See @ref onoff_release for more details.
+ *
+ * @param mgr the manager for which a request was successful.
+ *
+ * @retval non-negative the observed state (ONOFF_STATE_ON) of the
+ * machine at the time of the release, if the release succeeds.
+ * @retval negative on error returned by @ref onoff_release. See
+ * @ref onoff_release for details.
+ */
+static inline int onoff_delayed_release(struct onoff_delayed_manager *mgr)
+{
+	return onoff_release(&mgr->mgr);
+}
+
+/**
+ * @brief Attempt to cancel an in-progress client operation.
+ *
+ * See @ref onoff_cancel for details.
+ *
+ * @param mgr the manager for which an operation is to be cancelled.
+ *
+ * @param cli a pointer to the same client state that was provided
+ * when the operation to be cancelled was issued.
+ *
+ * @retval 0 if successfully cancelled before request delay has past.
```

Comment:
s/past/passed/

The text about "on successful cancellation" should probably be lifted up into the function description, since there are two distinct return value descriptions both of which have that effect.